src/NHibernate/Bytecode/Lightweight/ReflectionOptimizer.cs

```diff
@@ -132,6 +167,46 @@ private static void EmitCastToReference(ILGenerator il, System.Type type)
 		private static readonly MethodInfo GetterCallbackInvoke = ReflectHelper.GetMethod<GetterCallback>(
 			g => g.Invoke(null, 0));
 
+		private GetPropertyValueInvoker GenerateGetPropertyValueMethod(IGetter getter)
+		{
+			if (!(getter is IOptimizableGetter optimizableGetter))
+			{
+				return null;
```

Comment:
We should put the fallback right here rather than yielding `null`. It simplifies the code and aligns with `GenerateGetPropertyValuesMethod`, which also handles the fallback itself.